discovery/src/main/scala/akka/discovery/SimpleServiceDiscovery.scala

```diff
@@ -54,6 +55,29 @@ object SimpleServiceDiscovery {
         case None => host
       }
   }
+
+  /**
+   * Represents a service discovery lookup.
+   *
+   * Each mechanism should document what a Simple vs Full lookup means.
+   * Implementations are free to ignore port/protocol from a Full lookup
+   * but should not throw if they are set unexpectedly as this prevents the
+   * same query being sent to multiple mechanisms via the aggregate service
+   * discovery
+   */
+  @ApiMayChange
+  sealed abstract class Lookup {
+    def name: String
+  }
+
+  @ApiMayChange
+  final case class Simple(name: String) extends Lookup
+
+  /**
+   * The meaning of each field is up to the discovery mechanism.
+   */
+  @ApiMayChange
+  final case class Full(name: String, port: String, protocol: String) extends Lookup
```

Comment:
port more like portName perhaps?

Reply:
perhaps also `require(!=null)` on those? Document if null or empty is suggested etc